src/utils/session.ts

```diff
@@ -22,3 +22,26 @@ export function getPresenterAccountDetails(req: Request): Details {
 export function setPresenterAccountDetails(req: Request, details: Details) {
     req.session?.setExtraData(PRESENTER_ACCOUNT_SESSION_KEY, details);
 }
+
+export function generateUserDetails(req: Request): Details{
+    const signinInfo = req.session?.data?.signin_info;
+    const details = {} as Details;
+
+    details.email = signinInfo?.user_profile?.email as string;
+    details.userId = signinInfo?.user_profile?.id as string;
+
+    const forename = signinInfo?.user_profile?.forename as string;
+    const surname = signinInfo?.user_profile?.surname as string;
+
+    const name: Name = { forename, surname };
```

Comment:
The name object can be populate as an object literal sintead of using varaibles them putting those variable into the object.